compiler/rustc_passes/src/check_attr.rs

```diff
@@ -1961,20 +1961,35 @@ fn check_invalid_crate_level_attr(tcx: TyCtxt<'_>, attrs: &[Attribute]) {
         sym::automatically_derived,
         sym::start,
         sym::rustc_main,
+        sym::derive,
```

Comment:
Could you add `global_allocator`, `bench`, `test` and `test_case` to this list as well?
Also add a comment telling that these attributes are built-in attribute macros, so finding them in this position means that they weren't successfully resolved due to the "cannot determine resolution" errors, so we can report the suggestion.
Also add a FIXME telling to fix the "cannot determine resolution" error instead, and then remove the built-in macros from this list.

Reply:
And squash commits into one after that.

Reply:
Sure will do that all later today, do you know if it will ever actually be possible to resolve these as inner attributes in that position anyway? 

I assumed that because they can't be used in that position resolving them was impossible but I may be missing some key understanding of resolving.

Reply:
>do you know if it will ever actually be possible to resolve these as inner attributes in that position anyway?

Yes, if we skip the current module (and some prelude layers?) during name lookup.
In which context inner attributes should be resolved is a question in general (see https://github.com/rust-lang/rust/issues/54726 for some more details).

Reply:
Cool thanks I think I remember seeing the discussion about where we resolve inner attributes but it will be good to go back and read it with some better context. 

I suppose my thoughts are that attributes like derive almost don't do anything to a module (although maybe I'm wrong there) so I would expect something to fail, though I guess maybe that would be during expansion rather than resolution?

Reply:
Should be all sorted now - have added the other attributes into the test as well so it can be updated once inner attribute resolution is sorted

Reply:
>I guess maybe that would be during expansion rather than resolution?

Yes, the macro itself will check its target and report an error if necessary (`fn report_bad_target` in case of `derive`).